### ISSUE TITLE
Support "as" attribute on link elements

### ISSUE THREAD

**Issue by marcoscaceres** [htacg/tidy-html5#449]
Currently, 
```HTML 
    <link
      rel="preload"
      href="https://www.w3.org/scripts/whatevs.js"
      as="script">
```
Yields: 

```
Warning: <link> proprietary attribute "as"
```

However, `as` is a [standard](https://www.w3.org/TR/preload/) enumerated attribute, whose values are a [request destination](https://fetch.spec.whatwg.org/#concept-request-destination):

 * "report"
 * "document"
 * "document"
 * "object"
 * "embed"
 * "media"
 * "font"
 * "image"
 * "script"
 * "serviceworker"
 * "sharedworker"
 * "worker"
 * "style"
 * "media"
 * "media"
 * "image"
 * "manifest"
 * "xslt"

**Comment by marcoscaceres** [comment]
Note that it could be possible to simply support any string the value of `as`, as the interface for it is defined as returning a DOMString instead of an enum.

**Comment by geoffmcl** [comment]
@marcoscaceres thank you for the issue, and the PR fix #450... that is good to see - a problem, a link, and a solution at the same time...

Now my reading of the link - https://www.w3.org/TR/preload/ - I can see - 

````
W3C Working Draft 11 July 2016, 
and 
Implementers should be aware that this document is not stable....
````

So my question is `Should we update HTML Tidy at this time?`

It certainly seems the W3C Validation service has no problem with this attribute...

Maybe it could be implemented under a compile time switch of the form say `#ifdef IMPLEMENT_PRELOAD_AS` or something...

And then there is the question, should we support the value enumeration, or as suggested here - accept any string?

### But I strongly seek comments on what should be the Tidy update policy? 

Should Tidy try to follow **draft** W3C  specifications, like it seems the validator has done? 

Or should Tidy wait until the spec moves up the acceptance ladder?

### Need help to decide!

But again @marcoscaceres, thanks for bringing up the issue... also look forward to your further thoughts...

**Comment by marcoscaceres** [comment]
> And then there is the question, should we support the value enumeration, or as suggested here - accept any string?

I think we've done ok without this, but this would be nice to enable as a flag, as some attributes can have negative/unexpected effects when incorrectly set (e.g., if one makes a typo in iframe sandboxing attribute, that could be bad... potentially the same with the crossorigin attribute that can be used on various elements).  

> Should Tidy try to follow draft W3C specifications, like it seems the validator has done? 

Tl;dr: maybe it should be that if it's in two browsers, HTML validator supports it, and has positive signals from browser vendors, it qualifies. 

I'm of the opinion that we should support things that browsers implement. The `as` attribute is supported in both [Chrome and Opera](http://caniuse.com/#search=preload). 

It is also in active development in Firefox:
https://bugzilla.mozilla.org/show_bug.cgi?id=1222633

Last patch was submitted for review a few days ago.

> Or should Tidy wait until the spec moves up the acceptance ladder?

This is tricky, as it depends on what level we would want to use (e.g., some specs never reach REC, but are widely implemented - like WebIDL). And some standards bodies now completely avoid using an "acceptance ladder" (e.g., WHATWG) and even the W3C has started to move in that direction (e.g., with the [WICG](https://wicg.io/), which is the group I chair).  

Another option is to put things into an "experimental" or "stage" bucket, similarly to projects like Babble do: where you can enable certain features with a flag, but not require the flag at compile time. 

So, it would be nice to have something like:

```bash
# "experimental" is a misnomer... I'm sure we can think of something nicer
tidy --experimental -o output.html input.html
```

**Comment by geoffmcl** [comment]
@marcoscaceres thank you for your further feedback... I largely agree with you... especially about things **not** ever rising to `REC` status...

At this time I am inclined to accept your PR #450 as is. 

 1. It is just a `link` attribute, and has no html display consequences... 
 2. do not see the need to add an enum of values, but that could be added later, if desired...
 3. it does not seem worth adding another option, like `experimental` at this time for it...
 4. the W3C validator accepts it, regardless of actual browser support...

Baring any further negative comments I will merge this over the next few days... thanks...

**Comment by marcoscaceres** [comment]
Excellent, thank you @geoffmcl!
